### PR TITLE
HOTT-1295: Rails 7 with rails 6.1 configuration

### PR DESCRIPTION
**Pull request by willfish**
### Jira link

https://transformuk.atlassian.net/browse/HOTT-1295

### What?

I have added/removed/altered:

- [x] Bumped rails to 7.x
- [x] Moved to zeitwerk
- [x] Moved to ruby 6.1 configuration defaults

### Why?

I am doing this because:

- We're globally moving to rails 7 and ruby 3.1

### CODE REVIEW

File: config/initializers/assets.rb [hunk 1]
```diff
@@ -1,13 +1,10 @@
 # Be sure to restart your server when you modify this file.
 
 # Version of your assets, change this if you want to expire all your assets.
-Rails.application.config.assets.enabled = false
```
Comment (jebw) [inline config/initializers/assets.rb:4]: Dropping this line is probably why sprockets has re-appeared

**Review by jebw — CHANGES_REQUESTED** [review]
Looks good but I think this is bringing sprockets back in - I dropped this to take 4 minutes out the build time so should be kept.

File: config/initializers/assets.rb (hunk 1 above)
Comment (willfish) [reply config/initializers/assets.rb:4]: That's not quite correct. config.assets isn't a configuration section that is available without sprockets. This is disabled by the removal of sprockets from rails 7. I forgot you removed this and these are managed by webpacker

**Review by jebw — APPROVED** [review]
LGTM

**Pull request merged**